### ISSUE TITLE
Datetime and timedelta

### ISSUE THREAD

**Issue by jetheurer** [hgrecco/pint#510]
Can pint support determining units from a timedelta?  It would be useful if it could for dealing with raw data.

**Comment by hgrecco** [comment]
There is not such thing right now but it would be interesting.

**Comment by aglassdarkly** [comment]
Is this the use case?
```python
ureg = UnitRegistry()
t = ureg.Quantity(datetime.timedelta(days=4))
str(t)
'4.0 day'
```

**Comment by hgrecco** [comment]
I guess that is more in the direction of interoperating datetime objects with currently supported units such as:
```python
>>> # Currently possible
>>> 4 * ureg.hour
<Quantity(4, 'hour')>
>>> 4 * ureg.hour + 30 * ureg.min
<Quantity(4.5, 'hour')>
>>> # Not possible now
>>> datetime.datetime.now() + 4 * ureg.hour
```

**Comment by jetheurer** [comment]
Yes exactly.  I don't think its an easy problem to solve as there are a lot of nuances with dates and times, but as long it is was properly defined/documented it would be interesting.  python-dateutil has some interesting solutions.

**Comment by hgrecco** [comment]
I think we can start by providing a conversion utility between quantity and timedelta

**Comment by dopplershift** [comment]
Not sure if it's a good idea, but you could also put `isinstance` checks for date time in `__add__`, `__radd__`, etc.

**Comment by aglassdarkly** [comment]
I was exploring a bit and wrote a conversion between timedelta and quantity. I'll try hooking it up in the math ops and see what happens. This is an interesting issue!

**Comment by hgrecco** [comment]
I think that date rules are very similar to our offset units:
```
datetime - datetime = timedelta
datetime (+  or -) timedelta = datetime
timedelta (+  or -) timedelta = timedelta
```
and that Pint Quantities are conceptualy timedeltas. What do you think?

**Comment by aglassdarkly** [comment]
I agree that Pint quantities are conceptually timedeltas. Would it be as simple as adding something like this method to \_Quantity and calling it appropriately from \_\_add\_\_, \_\_sub\_\_, etc. when the other side is a datetime?
```python
    def to_timedelta(self):
        return datetime.timedelta(microseconds=self.to('microseconds').magnitude)
```

I can make a PR with these changes if this seems right.

**Comment by hgrecco** [comment]
Sounds good to me.

**Comment by hgrecco** [comment]
This is implemented in 0.8.1. Let's see how it plays out.

**Comment by Benjamin-Lee** [comment]
Can we get native support for converting timedeltas to quantities? 
```python
>>> ureg = UnitRegistry()
>>> t = ureg.Quantity(datetime.timedelta(days=4))
>>> str(t)
'4 days, 0:00:00 dimensionless'
>>> t  == ureg.parse_expression("4 seconds")
False
```
